File: Logic_HandsOn/C++/Problems on Digits,String,Array,Pointers etc/program21.cpp
/////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
//	Program:
//		a program which accept radius of circle from user and calculate its area.
//		Consider value of PI as 3.14. (Area = PI * Radius * Radius)
//
////////////////////////////////////////////////////////////////////////////////////////////////////////////////

#include<iostream>
using namespace std;

class Area
{
	private:
		float fRadius;

	public:

		Area(float fRadius)
		{
			this->fRadius = fRadius;
		}	

		double CircleArea(float PI=3.14,double dArea=0.0)
		{
			dArea = PI * fRadius * fRadius;
			return dArea;
		}
};

int main()
{
	float fValue = 0.0;
	double dRet = 0.0;

	cout<<"Enter radius";
	cin>>fValue;

	Area *aObj = new Area(fValue);
	dRet = aObj->CircleArea();

	cout<<"Area is "<<dRet<<"\n";

	return 0;
}

/*

	OUTPUT:

		Enter radius5.3
		Area is 88.2026

		Enter radius10.4
		Area is 339.6224

*/

File: Logic_HandsOn/Java/Problems on Array,Pattern etc/Program21.java
/////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
//	Program:
//		a program which accept radius of circle from user and calculate its area.
//		Consider value of PI as 3.14. (Area = PI * Radius * Radius)
//
////////////////////////////////////////////////////////////////////////////////////////////////////////////////

import java.util.*;

class Solution
{
	public float iValue;

	public Solution(float iValue)
	{
		this.iValue = iValue;
	}

	public double CircleArea()
	{

                double dArea = 3.14*iValue*iValue;      	
		return dArea;
	
	}
}

class Program
{
	public static void main(String[] args)
	{
		Scanner sObj = new Scanner(System.in);

		double dRet=0.0;

		System.out.println("Enter Radius: ");
		float fValue = sObj.nextFloat();

		Solution obj = new Solution(fValue);
		dRet = obj.CircleArea();
	
		System.out.println("Area is: "+dRet);
	}
}

/*
OUTPUT:

Enter Radius: 
5.3
Area is: 88.20260634841931

*/
